addons/web/static/src/core/model_field_selector/model_field_selector.js

```diff
@@ -89,6 +89,10 @@ export class ModelFieldSelector extends Component {
         }
         const { isInvalid, modelsInfo, names } = await this.fieldService.loadPath(resModel, path);
         const result = { isInvalid: !!isInvalid, displayNames: [] };
+        if (["properties", "properties_definition"].includes(modelsInfo.at(-1).type)) {
```

Comment:
not good. modelsInfo is an array of objects of the form { resModel, fieldDefs }. You could mayb check validity in loop below. There your have the fieldDef you want at last index